br-ext/package/optee_rust_ext/optee_rust_ext.mk

```diff
@@ -0,0 +1,46 @@
+OPTEE_RUST_EXT_VERSION = 1.0
+OPTEE_RUST_EXT_SOURCE = local
+OPTEE_RUST_EXT_SITE = $(BR2_PACKAGE_OPTEE_RUST_EXT_SITE)
+OPTEE_RUST_EXT_SITE_METHOD = local
+OPTEE_RUST_EXT_INSTALL_STAGING = YES
+
+EXAMPLE = $(wildcard examples/*)
+
+HOST_TARGET := aarch64-unknown-linux-gnu
+TA_TARGET := aarch64-unknown-optee-trustzone
+
+export RUST_TARGET_PATH=$(@D)
+export RUST_COMPILER_RT_ROOT=$(RUST_TARGET_PATH)/rust/rust/src/llvm-project/compiler-rt
+export OPTEE_DIR=$(@D)/../../..
+export OPTEE_OS_DIR=$(OPTEE_DIR)/optee_os
+export OPTEE_CLIENT_DIR=$(OPTEE_DIR)/out-br/build/optee_client_ext-1.0
+export OPTEE_CLIENT_INCLUDE=$(OPTEE_CLIENT_DIR)/out/export/usr/include
+export PATH+=:$(OPTEE_DIR)/toolchains/aarch64/bin
+export VENDOR=qemu_v8.mk
+export OPTEE_OS_INCLUDE=$(OPTEE_DIR)/optee_os/out/arm/export-ta_arm64/include
+
+define OPTEE_RUST_EXT_BUILD_CMDS
+	@$(foreach f,$(wildcard $(@D)/examples/*/Makefile), \
+		echo Building $f && \
+		$(MAKE) -C $(dir $f) &&) true
+endef
+
+define OPTEE_RUST_EXT_INSTALL_TARGET_CMDS
+	@$(foreach f,$(wildcard $(@D)/examples/*/ta/target/$(TA_TARGET)/release/*.ta), \
+		mkdir -p $(TARGET_DIR)/lib/optee_armtz && \
+		echo Installing $f && \
+		$(INSTALL) -v -p --mode=444 \
+			--target-directory=$(TARGET_DIR)/lib/optee_armtz $f \
+			&&) true
+	@$(foreach f,$(wildcard $(@D)/examples/*/host/target/$(HOST_TARGET)/release/*-rs), \
+		echo Installing $f && \
+		$(INSTALL) -v -p --target-directory=$(TARGET_DIR)/usr/bin $f \
```

Comment:
Here we're putting the Rust TA's in `/usr/bin/` instead of `/lib/optee_armtz/`. What is the motivation behind that? (I guess an upcoming patch in `optee_client` will look for Rust TA's there).

Reply:
Why do Rust TAs need a special place? I certainly hope they are loaded in the same way.

Reply:
This `foreach` puts the Rust CA's in `/usr/bin`.

Reply:
I see, thanks.